packages/core/src/external/cda/remove-b64.ts

```diff
@@ -37,18 +47,39 @@ export function removeBase64PdfEntries(payloadRaw: string): {
         if (comp.section.entry) {
           //eslint-disable-next-line @typescript-eslint/no-explicit-any
           comp.section.entry = toArray(comp.section.entry).filter((entry: any) => {
-            const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
-            if (
-              (BINARY_MIME_TYPES.includes(mediaType) || mediaType == undefined) &&
-              (entry.act?.text?.["_representation"]?.trim().toLowerCase() ===
-                b64Representation.toLowerCase() ||
-                entry.organizer?.component?.observationMedia?.value?.["_representation"]
-                  ?.trim()
-                  .toLowerCase() === b64Representation.toLowerCase())
-            ) {
-              isRemovedEntries = true;
-              b64Attachments.push(entry);
-              return false;
+            if (isConcernActEntry(entry)) {
+              const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
+              if (
+                (mediaType == undefined || BINARY_MIME_TYPES.includes(mediaType)) &&
+                isB64Representation(entry.act?.text?.["_representation"])
+              ) {
+                isRemovedEntries = true;
+                b64Attachments.acts.push(entry);
+                b64Attachments.total++;
+                return false;
+              }
+            } else if (isObservationOrganizer(entry)) {
+              const mediaComponent = getMediaObservations(entry);
+              // TODO: Apparently, some XML have B64 attachments in regular observations, so need to account for that as well
+              if (!mediaComponent?.length) return true;
+
+              let filterOut = false;
+              mediaComponent.map(comp => {
```

Comment:
wasn't this a `some` before -- I think there's an array method that will return a boolean for sure.

Reply:
my thinking here is that `some` might exit as soon as it finds a component that meets the criteria. But im not 100% sure there will always be just 1 component with b64 strings in it